DOC: Corrected Out-of-date list of sort algorithms

FullBeardDev opened the pull request:
Reference Issues/PRs
Fixes: #38287

Implemented changes:
Changed the docstring of all the methods using np.sort as it now includes one additional stable algorithm

Commit:
DOC: Corrected Out-of-date list of sort algorithms

Reference Issues/PRs
Fixes: #38287

Implemented changes:
Changed the docstring of all the methods using np.sort as it now includes one additional stable algorithm

jreback commented:
can you also update the tests in test_nargsort to add 'stable' in pandas/tests/test_sorting.py

Commit:
TST: Added the new supported sorting algorithm stable to test

Addition to doc Fix PR: #38503

Implemented changes:
Parameterized tests with addition of the newly supported sorting method stable from numpy 1.15.0

DonPablo99 commented:
> can you also update the tests in pandas/tests/series/methods/test_sort_index.py to use stable (where we use mergesort), just parameterize a test is prob good enough.

Done :)

mroeschke commented:
LGTM, mind fixing up the merge conflict @DonPablo99?

jreback commented on pandas/tests/series/methods/test_sort_index.py:
```diff
@@ -104,18 +104,13 @@ def test_sort_index_multiindex(self, level):
         tm.assert_series_equal(s, res)
 
-    def test_sort_index_kind(self):
+    @pytest.mark.parametrize("kind", ["quicksort", "mergesort", "heapsort", "stable"])
```
can you make this a fixture (put in the top of this test file is fine) as well.

jreback replied:
can you do this

DonPablo99 replied:
Should I just add a fixture or add it and replace the parametrize statements ?

jreback replied:
add a fixture in this file and use it

jreback commented:
can you merge master and update

jreback reviewed (changes requested):
comment on testing. ping on green.

jreback commented:
thanks @DonPablo99
